weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+    elif enforce_high_volume and not on_compute_engine:
+        raise RuntimeError(
+            'Must run on a compute engine VM to use the high volume earth engine api.'
+        )
+    elif enforce_high_volume and not up_to_date:
+        raise RuntimeError(
+            'Earth engine api must be updated to version ≥0.1.263 to use the high volume api'
+        )
+
+    # Ensure the api version is acceptable
+    elif not up_to_date and not allow_old_versions:
+        raise RuntimeError(
+            'Earth engine api is out of date. Update to version ≥0.1.263.')
+
+    # Try using the service account
+    elif on_compute_engine:
+        ee.Initialize(creds)
+
+    # Try using an personal authentication
+    else:
+        ee.Initialize()
+
+
+@dataclasses.dataclass
+class GribData(object):
+    """A class for holding the output of the GetChannelsDoFn."""
+    name: str  # The EE-safe name of the channel.
+    start_time: float  # In floating point seconds since epoch.
+    end_time: float  # In floating point seconds since epoch.
+    dtype: str  # data type of the channel data
+    projection: str  # The gdal string defining the projection.
+    transform: t.Tuple[float, float, float, float, float, float]  # From gdal.
+    metadata: t.Dict[str, t.Union[str, float, int]]
+    data: t.List[xr.Dataset]  # The list of datasets
+
+
+@dataclasses.dataclass
+class ToEarthEngine(ToDataSink):
+    """Loads weather data into Google Earth Engine.
+
+    TODO(alxr): Document...
+    """
+    example_uri: str
+    xarray_open_dataset_kwargs: t.Dict
+    disable_in_memory_copy: bool
+    disable_grib_schema_normalization: bool
+    skip_region_validation: bool
+    ee_asset: str
+    tiff_location: str
+    use_personal_account: bool
+
+    @classmethod
+    def add_parser_arguments(cls, subparser: argparse.ArgumentParser):
+        subparser.add_argument('--xarray_open_dataset_kwargs', type=json.loads, default='{}',
+                               help='Keyword-args to pass into `xarray.open_dataset()` in the form of a JSON string.')
+        subparser.add_argument('--disable_in_memory_copy', action='store_true', default=False,
+                               help='To disable in-memory copying of dataset. Default: False')
+        subparser.add_argument('--disable_grib_schema_normalization', action='store_true', default=False,
+                               help='To disable merge of grib datasets. Default: False')
+        subparser.add_argument('--ee_asset', type=str, default=None,
+                               help='The asset folder path in earth engine project where the tiff image files'
+                               ' will be pushed.')
+        subparser.add_argument('-s', '--skip-region-validation', action='store_true', default=False,
+                               help='Skip validation of regions for data migration. Default: off')
+        subparser.add_argument('--tiff_location', type=str, default=None,
```

Comment:
Please mark the required arguments as required, i.e.: https://docs.python.org/3/library/argparse.html#required

It's better to check this via argparse then via assert later on. 

Last, let's move all the required arguments to the to of this method.